src/main/java/se/sundsvall/casedata/service/ErrandService.java

```diff
@@ -335,6 +336,26 @@ public void replaceStatusesOnErrand(final Long id, final List<StatusDTO> dtos) {
 		processService.updateProcess(updatedErrand);
 	}
 
+	@Retry(name = "OptimisticLocking")
+	public void replaceFacilitiesOnErrand(final Long id, final List<FacilityDTO> dtos) {
+		final var oldErrand = getErrand(id);
+		final var facilitiesToChange = oldErrand.getFacilities().stream().filter(facility -> dtos.stream().map(FacilityDTO::getId).toList().contains(facility.getId())).toList();
+		final var newFacilities = dtos.stream().filter(dto -> !facilitiesToChange.stream().map(Facility::getId).toList().contains(dto.getId())).map(EntityMapper::toFacility).toList();
+
+		oldErrand.getFacilities().removeAll(facilitiesToChange);
```

Comment:
A PUT should replace everything. This will leave elements that are not included in the request, right?

Reply:
I thinked about this after work yesterday and was going to change this. I'll fix.

Reply:
But it's still al lot of strange code in the method?

Shouldn't this method just: 
1. Map the FacilityDTO-objects to FacilitiyEntity-objects. 
2. Put these objects in the facilitiesArray (i.e replace all other stuff, if any) 
3. Save

Reply:
Have a look at ErrandService.replaceStakeholdersOnErrand.